mcda/utils/utils_for_main.py

```diff
@@ -429,14 +470,30 @@ def save_df(df: pd.DataFrame, folder_path: str, filename: str):
     timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
     new_filename = f"{timestamp}_{filename}"
 
-    result_path = os.path.join(folder_path, new_filename)
-    df.to_csv(path_or_buf=result_path, index=False)
+    if not os.path.isdir(folder_path):
+        logging.error(f"The provided folder path '{folder_path}' is not a valid directory.")
+        return
+
+    full_output_path = os.path.join(output_directory_path, folder_path, new_filename)
+
+    try:
+        ensure_directory_exists(os.path.dirname(full_output_path))
+    except Exception as e:
+        logging.error(f"Error while saving data frame: {e}")
```

Comment:
I believe that only this check is enough. The one above is not required.

Reply:
They don't harm for now, I'll leave them.